messages/message_manager.go

```diff
@@ -6,63 +6,37 @@
 package messages
 
 import (
-	"fmt"
-
 	"github.com/ava-labs/avalanchego/ids"
-	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
-	"github.com/ava-labs/awm-relayer/config"
-	offchainregistry "github.com/ava-labs/awm-relayer/messages/off-chain-registry"
-	"github.com/ava-labs/awm-relayer/messages/teleporter"
-	"github.com/ava-labs/awm-relayer/vms"
 	"github.com/ethereum/go-ethereum/common"
 )
 
 // MessageManager is specific to each message protocol. The interface handles choosing which messages to send
 // for each message protocol, and performs the sending to the destination chain.
 type MessageManager interface {
+	// Create a message handler to relay the Warp message
+	NewMessageHandler(unsignedMessage *warp.UnsignedMessage) (MessageHandler, error)
+}
+
+// MessageHandlers relay a single Warp message. A new instance should be created for each Warp message.
+type MessageHandler interface {
```

Comment:
I'm a bit confused what the need is to separate `MessageManager` and `MessageHandler` here, and worried it could easily be confused what each is intended to be used for in the future. 

It seems like a `MessageManager` is just a wrapper for creating `MessageHandler`s. Would it be possible to remove it all together and just use `NewMessageHandler` directly with additional params as needed? Otherwise, I think we should consider renaming `MessageManager` to `MessageHandlerFactory` or something similar.

Reply:
I renamed `MessageManager` to `MessageHandlerFactory`, and put both it and `MessageHandler` in `message_handler.go`.